From c20196f7b1dde6c6ad3489a74e03780f011a7817 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:43 -0400
Subject: [cpufreq] powernow: don't initialize part of policy set by core

Message-id: <1409235879-1283-70-git-send-email-prarit@redhat.com>
Patchwork-id: 89387
O-Subject: [RHEL7.1 PATCH BZ 1134639 069/185] cpufreq: powernow: don't initialize part of policy set by core
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit eaf8120e8f9c4efc71475b1e08b8e2d3e421e6ca
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:28:55 2013 +0530

    cpufreq: powernow: don't initialize part of policy set by core

    Many common initializations of struct policy are moved to core now and hence
    this driver doesn't need to do it. This patch removes such code.

    Most recent of those changes is to call ->get() in the core after calling
    ->init().

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/powernow-k8.c b/drivers/cpufreq/powernow-k8.c
index 2efb759..85d60e3 100644
--- a/drivers/cpufreq/powernow-k8.c
+++ b/drivers/cpufreq/powernow-k8.c
@@ -1141,9 +1141,6 @@ static int powernowk8_cpu_init(struct cpufreq_policy *pol)
  cpumask_copy(pol->cpus, cpu_core_mask(pol->cpu));
  data->available_cores = pol->cpus;
 
- pol->cur = find_khz_freq_from_fid(data->currfid);
- pr_debug("policy current frequency %d kHz\n", pol->cur);
-
  /* min/max the cpu is capable of */
  if (cpufreq_table_validate_and_show(pol, data->powernow_table)) {
   printk(KERN_ERR FW_BUG PFX "invalid powernow_table\n");
-- 
1.7.1